spec/model/github/github_repository_spec.rb

```diff
@@ -13,25 +13,44 @@
-    github_repository.destroy_blob_storage
+  describe ".destroy_blob_storage" do
+    it "deletes the bucket and token" do
+      expect(blob_storage_client).to receive(:delete_bucket).with(bucket: "gph0hh0ahdbj6ng2cc3rvdecr8")
+      expect(cloudflare_client).to receive(:delete_token).with(github_repository.access_key)
+      expect(github_repository).to receive(:update).with(access_key: nil, secret_key: nil)
+      github_repository.destroy_blob_storage
+    end
+
+    it "runs even if the bucket already deleted" do
+      expect(blob_storage_client).to receive(:delete_bucket).and_raise(Aws::S3::Errors::NoSuchBucket.new(nil, nil))
+      expect(cloudflare_client).to receive(:delete_token).with(github_repository.access_key)
+      expect(github_repository).to receive(:update).with(access_key: nil, secret_key: nil)
+      github_repository.destroy_blob_storage
+    end
   end
 
-  it ".setup_blob_storage" do
-    expect(Config).to receive_messages(github_cache_blob_storage_region: "weur", github_cache_blob_storage_account_id: "123")
-    expect(blob_storage_client).to receive(:create_bucket).with({bucket: "gph0hh0ahdbj6ng2cc3rvdecr8", create_bucket_configuration: {location_constraint: "weur"}})
-    expected_policy = [
-      {
-        "effect" => "allow",
-        "permission_groups" => [{"id" => "2efd5506f9c8494dacb1fa10a3e7d5b6", "name" => "Workers R2 Storage Bucket Item Write"}],
-        "resources" => {"com.cloudflare.edge.r2.bucket.123_default_gph0hh0ahdbj6ng2cc3rvdecr8" => "*"}
-      }
-    ]
-    expect(cloudflare_client).to receive(:create_token).with("gph0hh0ahdbj6ng2cc3rvdecr8-token", expected_policy).and_return(["test-key", "test-secret"])
-    expect(github_repository).to receive(:update).with(access_key: "test-key", secret_key: Digest::SHA256.hexdigest("test-secret"))
-    github_repository.setup_blob_storage
+  describe ".setup_blob_storage" do
+    it "creates a bucket and token" do
+      expect(Config).to receive_messages(github_cache_blob_storage_region: "weur", github_cache_blob_storage_account_id: "123")
+      expect(blob_storage_client).to receive(:create_bucket).with({bucket: "gph0hh0ahdbj6ng2cc3rvdecr8", create_bucket_configuration: {location_constraint: "weur"}})
+      expected_policy = [
+        {
+          "effect" => "allow",
+          "permission_groups" => [{"id" => "2efd5506f9c8494dacb1fa10a3e7d5b6", "name" => "Workers R2 Storage Bucket Item Write"}],
+          "resources" => {"com.cloudflare.edge.r2.bucket.123_default_gph0hh0ahdbj6ng2cc3rvdecr8" => "*"}
+        }
+      ]
+      expect(cloudflare_client).to receive(:create_token).with("gph0hh0ahdbj6ng2cc3rvdecr8-token", expected_policy).and_return(["test-key", "test-secret"])
+      expect(github_repository).to receive(:update).with(access_key: "test-key", secret_key: Digest::SHA256.hexdigest("test-secret"))
+      github_repository.setup_blob_storage
+    end
+
+    it "runs even if the bucket already exists" do
```

Comment:
maybe: succeeds if the bucket already exists

Reply:
Thanks. Changed